examples/react-full/renderer/pageContext.d.ts

```diff
@@ -2,22 +2,18 @@ import 'vite-plugin-ssr';
 
 declare module "vite-plugin-ssr" {
   namespace VitePluginSsr {
-    interface PageContextInit {
-      pageExports: {
-        documentProps?: {
-          title: string
-        }
-      }
-    }
+    interface PageContextInit {}
 
     // Is it possible to also support `pageContext` on the browser-side by doing something like the following?
     interface PassToClient {
       value: 'pageProps' | 'documentProps'
     }
 
     interface PageContextOnBeforeRender {
-      documentProps?: {
-        title: string
+      pageContext?: {
```

Comment:
I added `pageContext` here so that it matches what I defined in the types.ts files, but we could also auto wrap `PageContextOnBeforeRender` into a `{ pageContext: PageContextOnBeforeRender }` in types.ts. It just a matter of letting this responsibility to the user or not

Reply:
If it's possible then yes; other than `pageContext` I don't forsee any other properties.

So I guess we can straight for:

```ts
interface PageContextOnBeforeRender {
  redirectUrl?: string
}
```

Reply:
Although, I'm thinking, how would the user assign everything at once?

```ts
interface PageContextOnBeforeRender {
  // Is this possible if we don't have the `pageContext` indirection? 
  pageContext: PageContextTypeDefinedByUser
}
```

Or maybe we do this:

```ts
// This naming seems nice as well
interface OnBeforeRender {
  pageContext: PageContextTypeDefinedByUser
}
```

Reply:
> ```ts
> // This naming seems nice as well
> interface OnBeforeRender {
>   pageContext: PageContextTypeDefinedByUser
> }
> ```

That's a good idea, the interface name not containing `PageContext` and having a `pageContext` is clearer I think, and it's future proof if we want to return something other then `pageContext`

Reply:
renamed in 1166de1